airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -0,0 +1,314 @@
+    are used as credentials usually and you can specify different authentication parameters
+    via init params or via corresponding extras in the connection.
+
+    The extras in the connection are named the same as the parameters (`mount_point`,'kv_engine_version' ...).
+
+    Login/Password are used as credentials:
+
+        * approle: password -> secret_id
+        * aws_iam: login -> key_id, password -> secret_id
+        * azure: login -> client_id, password -> client_secret
+        * ldap: login -> username,   password -> password
+        * userpass: login -> username, password -> password
+        * radius: password -> radius_secret
+
+    :param vault_conn_id: The id of the connection to use
+    :type vault_conn_id: str
+    :param auth_type: Authentication Type for Vault. Default is ``token``. Available values are in
+        :py:const:`airflow.providers.hashicorp.common.vault_client.VALID_AUTH_TYPES`.
+    :type auth_type: str
+    :param mount_point: The "path" the secret engine was mounted on. Default depends on the engine used.
+    :type mount_point: str
+    :param kv_engine_version: Select the version of the engine to run (``1`` or ``2``). Default 2.
+    :type kv_engine_version: int
+    :param token: Authentication token to include in requests sent to Vault.
+        (for ``token`` and ``github`` auth_type)
+    :type token: str
+    :param role_id: Role ID for Authentication (for ``approle``, ``aws_iam`` auth_types)
+    :type role_id: str
+    :param kubernetes_role: Role for Authentication (for ``kubernetes`` auth_type)
+    :type kubernetes_role: str
+    :param kubernetes_jwt_path: Path for kubernetes jwt token (for ``kubernetes`` auth_type, default:
+        ``/var/run/secrets/kubernetes.io/serviceaccount/token``)
+    :type kubernetes_jwt_path: str
+    :param gcp_key_path: Path to GCP Credential JSON file (for ``gcp`` auth_type)
+           Mutually exclusive with gcp_keyfile_dict
+    :type gcp_key_path: str
+    :param gcp_keyfile_dict: Dictionary of keyfile parameters. (for ``gcp`` auth_type).
+           Mutually exclusive with gcp_key_path
+    :type gcp_keyfile_dict: dict
+    :param gcp_scopes: Comma-separated string containing GCP scopes (for ``gcp`` auth_type)
+    :type gcp_scopes: str
+    :param azure_tenant_id: Tenant of azure (for ``azure`` auth_type)
+    :type azure_tenant_id: str
+    :param azure_resource: Resource if of azure (for ``azure`` auth_type)
+    :type azure_resource: str
+    :param radius_host: Host for radius (for ``radius`` auth_type)
+    :type radius_host: str
+    :param radius_port: Port for radius (for ``radius`` auth_type)
+    :type radius_port: int
+
+    """
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        vault_conn_id: str,
+        auth_type: Optional[str] = None,
+        mount_point: Optional[str] = None,
+        kv_engine_version: Optional[int] = None,
```

Comment:
```suggestion
        kv_engine_version: Optional[int] = DEFAULT_KV_ENGINE_VERSION,
```

Based on comment on L122

Reply:
I fixed the description. It was misleading indeed:

* Hook's constructor kv_engine_version overrides whatever we have in connection. If we set anything there - so be it.
* If we did not set anything in the connection (or set it to None) we check the connection one and use what we find there. If there is nothing in the connection we default to 2. 

We cannot have default in the constructor otherwise we would not be able to distinguish "not set" from "set to 2".